src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java

```diff
@@ -195,20 +208,44 @@ public File getEventsFile() {
         return new File(folder.getComputationDir(), "events.log");
     }
 
-    public TaskListener createEventsListener() throws IOException {
+    @WithBridgeMethods(TaskListener.class)
+    @Nonnull
+    public synchronized StreamTaskListener createEventsListener() {
         File eventsFile = getEventsFile();
-        FileUtils.forceMkdir(eventsFile.getParentFile());
-        boolean rotate = eventsFile.length() > EVENT_LOG_MAX_SIZE * 1024;
-        OutputStream os;
-        if (BACKUP_LOG_COUNT != null) {
-            os = new ReopenableRotatingFileOutputStream(eventsFile, BACKUP_LOG_COUNT);
-            if (rotate) {
-                ((ReopenableRotatingFileOutputStream) os).rewind();
-            }
-        } else {
-            os = new FileOutputStream(eventsFile, !rotate);
+        if (!eventsFile.getParentFile().isDirectory() && !eventsFile.getParentFile().mkdirs()) {
+            LOGGER.log(Level.WARNING, "Could not create directory {0} for {1}",
+                    new Object[]{eventsFile.getParentFile(), folder.getFullName()});
+            // TODO return a StreamTaskListener sending output to a log, for now this will just try and fail to write
+        }
+        if (eventStreams == null) {
+            eventStreams = new EventOutputStreams(new EventOutputStreams.OutputFile() {
+                @NonNull
+                @Override
+                public File get() {
+                    return getEventsFile();
+                }
+
+                @Override
+                public boolean canWriteNow() {
+                    GregorianCalendar timestamp = new GregorianCalendar();
+                    timestamp.setTimeInMillis(System.currentTimeMillis() - 10000L);
+                    Queue.Item probe = new Queue.WaitingItem(timestamp, folder, Collections.<Action>emptyList());
+                    for (QueueTaskDispatcher d: QueueTaskDispatcher.all()) {
+                        if (d.canRun(probe) != null) {
```

Comment:
Ooooh interesting question there... certainly when the item is one on the queue and not a probe instance that will never go next nor near the queue... we would need the lock... I'll have to mull this over, but my initial feeling is no... if it is not safe I may need to add a mix-in interface and adjust folders-plus